src/Lean/Data/Position.lean

```diff
@@ -29,9 +29,16 @@ instance : ToExpr Position where
 
 end Position
 
+/-- Content of a file together with precalculated positions of newlines. -/
 structure FileMap where
+  /-- The content of the file. -/
   source    : String
+  /-- The positions of newline characters.
+  The first entry is always `0` and the last always the index of the last character.
+  In particular, if the last character is a newline, that index will appear twice. -/
   positions : Array String.Pos
+  /-- The line indices. Has the same length as `positions` and is always of the form
+  `[1, 2, …, n-1, n-1]`. -/
```

Comment:
I'm not sure I understand this specification here. There's a few things about it that I find confusing:

1. (minor) It uses list syntax for an array
2. I'm not sure what a line index is. Can you define the term? Or what about calling it "line number associated with each mapped newline"?

Reply:
updated!

One point is that `lines` seems to be pretty redundant, as far as I can tell.

Reply:
Yes, if you want to do a follow-up PR removing it, that would be a strict improvement I'd say!